python/tests/test_scip.py

```diff
@@ -42,3 +42,122 @@ def test_set_param(model, name, param_type):
         value = param_type(1)  # Cast one to the required type
     model.set_param(name, value)
     assert model.get_param(name) == value
+
+
+@pytest.mark.parametrize("name,param_type", names_types)
+def test_get_set_param(model, name, param_type):
+    value = model.get_param(name)
+    model.set_param(name, value)
+    assert model.get_param(name) == value
+
+
+should_fail = (
+    # Bool in [TRUE,FALSE]
+    ("branching/preferbinary", -1),
+    ("branching/preferbinary", 0.1),
+    ("branching/preferbinary", "toto"),
+    ("branching/preferbinary", "a"),
+    ("branching/preferbinary", ""),
+    ("branching/preferbinary", None),
+    # Int in [-1,2147483647]
+    ("conflict/maxlploops", -5),
+    ("conflict/maxlploops", 1e10),
+    ("conflict/maxlploops", 0.1),
+    ("conflict/maxlploops", "toto"),
+    ("conflict/maxlploops", "a"),
+    ("conflict/maxlploops", ""),
+    ("conflict/maxlploops", None),
+    # Long int in [-1,9223372036854775807]
+    ("limits/nodes", -5),
+    ("limits/nodes", 1e20),
+    ("limits/nodes", 0.1),
+    ("limits/nodes", "toto"),
+    ("limits/nodes", "a"),
+    ("limits/nodes", ""),
+    ("limits/nodes", None),
+    # Real in [0,1e+20]
+    ("limits/time", -5),
+    ("limits/time", -0.1),
+    ("limits/time", 1e21),
+    ("limits/time", "toto"),
+    ("limits/time", "a"),
+    ("limits/time", ""),
+    ("limits/time", None),
+    # Real in [0,1.79769313486232e+308]
+    ("heuristics/undercover/maxcoversizeconss", -5),
+    ("heuristics/undercover/maxcoversizeconss", -0.1),
+    ("heuristics/undercover/maxcoversizeconss", "toto"),
+    ("heuristics/undercover/maxcoversizeconss", "a"),
+    ("heuristics/undercover/maxcoversizeconss", ""),
+    ("heuristics/undercover/maxcoversizeconss", None),
+    # Char in [d,s,u,v]
+    # ("branching/pscost/strategy", 0), NUL character valid for SCIP ?
+    ("branching/pscost/strategy", 6),
+    ("branching/pscost/strategy", -5),
+    ("branching/pscost/strategy", -0.1),
+    ("branching/pscost/strategy", "toto"),
+    ("branching/pscost/strategy", "a"),
+    ("branching/pscost/strategy", ""),
+    ("branching/pscost/strategy", True),
+    # ("branching/pscost/strategy", False), NUL character valid for SCIP ?
+    ("branching/pscost/strategy", None),
+    # String, combination of 'l', 'n', 'i' characters
+    ("heuristics/undercover/fixingalts", -5),
+    ("heuristics/undercover/fixingalts", -0.1),
+    ("heuristics/undercover/fixingalts", 6),
+    # ("heuristics/undercover/fixingalts", "toto"),  valid for SCIP ?
+    # ("heuristics/undercover/fixingalts", "a"),  valid for SCIP ?
+    ("heuristics/undercover/fixingalts", True),
+    ("heuristics/undercover/fixingalts", False),
+    ("heuristics/undercover/fixingalts", None),
+)
+
+
+@pytest.mark.parametrize("name,value", should_fail)
+def test_set_param_fail(model, name, value):
+    with pytest.raises((ecole.scip.Exception, TypeError)):
```

Comment:
Are we ever getting `TypeError`? The method should _apriori_ accept all these types, and `Model` is deciding whether it is valid, no?

Reply:
Pybind will raise `TypeError`s, for example with `None`s. Pybind is responsible for checking and casting types, I think it is legitimate to use it as much as possible, as it does a pretty good job at it.

Reply:
I agree, `None` should be processed by pybind.